core/src/main/java/smile/classification/DecisionTree.java

```diff
@@ -654,20 +704,26 @@ public Node findBestSplit(int n, int low, int high, int[] count, int[] falseCoun
                     if (gain > splitNode.splitScore) {
                         // new best split
                         splitNode.splitFeature = j;
-                        splitNode.splitValue = (x[o][j] + prevx) / 2;
+                        splitNode.splitValue = (x_oj + prevx) / 2;
                         splitNode.splitScore = gain;
                         splitNode.trueChildOutput = Math.whichMax(trueCount);
                         splitNode.falseChildOutput = Math.whichMax(falseCount);
                     }
 
-                    prevx = x[o][j];
-                    prevy = y[o];
-                    trueCount[y[o]] += samples[o];
+                    prevx = x_oj;
+                    prevy = y_o;
+                    trueCount[y_o] += samples[o];
                 }
             } else {
                 throw new IllegalStateException("Unsupported attribute type: " + attributes[j].getType());
             }
 
+            if(replaceCount + (countNaN > 0 ? 1: 0) <= 1) {
+                synchronized (constFeatures) {
+                    this.constFeatures = SmileUtils.sortedArraySet(constFeatures, j);
```

Comment:
I found this line confusing, since it's referring to the same thing (this.constFeatures) by two different names (this.constFeatures, constFeatures) - leading me to wonder if they were two different entities at this point.

Reply:
Fixed in https://github.com/haifengl/smile/pull/461/commits/74ce0f768deba72f72ab55488911f9c365b458b8